dynamax/utils.py

```diff
@@ -87,4 +84,34 @@ def random_rotation(seed, n, theta=None):
     out = jnp.eye(n)
     out = out.at[:2, :2].set(rot)
     q = jnp.linalg.qr(jr.uniform(key2, shape=(n, n)))[0]
-    return q.dot(out).dot(q.T)
\ No newline at end of file
+    return q.dot(out).dot(q.T)
+
+
+def add_batch_dim(tree, instance_shapes):
+    """Add a batch dimension to an array, if necessary.
```

Comment:
Technically speaking, this should be `possibly_add_batch_dim`, but I prefer your shorter name :)

Reply:
now `ensure_has_batch_dim` :)

Reply:
I like that